lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -109,6 +112,74 @@ def get(self, request, course_id):
         return Response(get_course(request, course_key))
 
 
+class CourseUsersView(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Fetch statistics about a user's activity in a course.
+
+    **Example Requests**:
+
+        GET /api/discussion/v1/users/?course_id=course-v1:ExampleX+Subject101+2015&order_by=activity
+
+    **Response Values**:
+
+
+    **Example Response**
+    ```json
+    {
+        "pagination": {
+            "count": 3,
+            "next": null,
+            "num_pages": 1,
+            "previous": null
+        },
+        "results": [
+            {
+                "active_flags": 3,
+                "inactive_flags": 0,
+                "replies": 13,
+                "responses": 21,
+                "threads": 32,
+                "username": "edx"
```

Comment:
question: I think the data also includes `author_id` here?

Reply:
Not here, here it's converted to "username", the id isn't part of the API response.

Reply:
quick question: we don't store user info in cs_comments_service mongo db?  e.g. I can see in the mongo model, there is a field `author_username` - can't that be used instead of `author_id`?
 See: https://github.com/edx/cs_comments_service/blob/master/models/content.rb#L8

Reply:
I don't think the author_username is indexed in mongo so I couldn't query based on that, but I'll see if I can somehow include that data form the mongo side, since it might save a query here.

Reply:
I looked into this, and it seems like it's possible. AFAIK the platform doesn't currently provide a way to change usernames, so it should work. I can copy through the username during aggregation, and provide that back from the ruby API so we don't need the lookup here. 
It could have issues if a user's username ever changes. I'll add a PR for that soon.

Reply:
I've updated this PR to use the usernames provided by cs_comments_service.

Reply:
Great, I'll take a look.